client/src/features/groupsV2/search/GroupSearchQueryInput.tsx

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright 2025 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import cx from "classnames";
+import { useCallback, useEffect } from "react";
+import { Search, XCircleFill } from "react-bootstrap-icons";
+import { Controller, useForm } from "react-hook-form";
+import { useSearchParams } from "react-router";
+import { Button, Form, InputGroup } from "reactstrap";
+import { FILTER_PAGE, FILTER_QUERY } from "./groupsSearch.constants";
+
+interface SearchBarForm {
+  query: string;
+}
+export default function GroupSearchQueryInput() {
+  // Set the input properly
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get(FILTER_QUERY.name) ?? "";
+
+  const { control, register, handleSubmit, setFocus } = useForm<SearchBarForm>({
+    defaultValues: { query },
+  });
+
+  // focus search input when loading the component
+  useEffect(() => {
+    setFocus("query");
+  }, [setFocus]);
+
+  const onClick = useCallback(() => {
```

Comment:
To make this action have an explicit name (clicking the clear button):
```suggestion
  const onClear = useCallback(() => {
```